### PR TITLE
Python3 support

### PR DESCRIPTION
**Pull request by YannickDieter**
This PR adds Python3 support (Python2 is still supported) for pyTLU.
Concerning dependencies of pyTLU, basil 3.0.0 or higher is now required and also the latest version of the online monitor is required (0.4.1).

Both python versions 2 and 3 are tested with travis.

### CODE REVIEW

**Comment by themperek** [comment]
This is ready ?

**Comment by YannickDieter** [comment]
From my side, this would be ready.

File: tests/test_Sim.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def check_data(self, how_many, tdc_en=False, start=0):
         if tdc_en:
             exp_tdc = np.array([135] * how_many, dtype=np.uint32)
             ret_tdc = ret[~tlu_word] >> 20
-            self.assertEqual((ret_tdc / 2).tolist(), (exp_tdc / 2).tolist())
+            self.assertEqual((ret_tdc // 2).tolist(), (exp_tdc // 2).tolist())
```
Comment (DavidLP) [inline tests/test_Sim.py]: Why not:
```python
self.assertEqual(ret_tdc // 2, exp_tdc // 2)
```

It does the same though ;-)

File: .travis.yml [hunk 2]
```diff
@@ -18,20 +19,23 @@ install:
...
-  - conda install --yes pytest numpy psutil qtpy pyqt pyyaml pyzmq pytables
-  - pip install cocotb
+  - conda create --yes -n test-environment python=$TRAVIS_PYTHON_VERSION pytest numpy psutil qtpy pyqt pyyaml pyzmq pytables
+  - source activate test-environment
+  - pip install cocotb==1.0.dev3
   - pip install xvfbwrapper  # fake x server for Qt gui tests
 
   # Install basil
-  - git clone -b v2.4.12 https://github.com/SiLab-Bonn/basil; cd basil; python setup.py develop; cd ..;
+  - git clone -b v3.0.0 https://github.com/SiLab-Bonn/basil; cd basil; python setup.py develop; cd ..;
```
Comment (themperek) [inline .travis.yml]: We should be able to pool just from pip now.

Comment (YannickDieter) [reply .travis.yml]: Yes, basil-daq 3.0.0 is available with pip.

File: tests/test_Sim.py (hunk 1 above)
Comment (YannickDieter) [reply tests/test_Sim.py]: This gives ``ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()``. Apparently ``self.assertEqual`` works only with lists. One could use
``` python
np.testing.assert_array_equal(ret_tdc // 2, exp_tdc // 2)
```
in order to check directly two arrays..

**Comment by themperek** [comment]
I can marge. Should I squash? @YannickDieter

**Comment by YannickDieter** [comment]
Thanks. Yes, squashing the commits would be a good idea.

**Pull request merged**